licence-checker/licence-checker.py

```diff
@@ -52,6 +53,20 @@ def search_line(self, licence_first_word):
     def expected_full_line(self, licence_line):
         return self.comment_prefix + ' ' + licence_line
 
+    def match(self, licence, line, match_regex):
+        """Match a license line wih a retrieved line using either string matching or regex
+        given the comment style.
+        """
+        if match_regex:
+            return re.compile(self.expected_full_line(licence)).match(line)
+        else:
+            return line == self.expected_full_line(licence)
+
+
+class LineCommentStyle(CommentStyle):
+    """Helpers for line-style comments."""
+    pass
```

Comment:
The inheritance looks right here, but can you move the body of `expected_full_line` here from `CommentStyle`, then define a version in the base class that just raises a `NotImplementedError`? That's closer to how things worked before (where we had two duck-typed classes that defined different versions of `expected_full_line`).